### DISCUSSION TITLE
rtmp from veriskope

### DISCUSSION THREAD

**Discussion by ciaoamigoschat** [ruffle-rs/ruffle#2151] [General]
https://github.com/veriskope/rtmp-rs

**Comment by grishka** [comment]
Networking is the one thing you won't be able to 100% replicate in a wasm flash player because of the differing security models in browsers vs flash plugin. RTMP in particular is a raw TCP socket and the cross-origin permissions are enforced by the server iirc after receiving the initial `connect` command. You could probably put an RTMP server behind some sort of websocket-terminating proxy to make it work. So while one could, in principle, make use of this library, it would only work in desktop builds.

**Comment by ROBERT-MCDOWELL** [comment]
I don't agree. with HTTP3 (webtransport) you can create any kind of TCP and UDP protcols, secured or not. also
tunneling can be done when you control low level socket layers.
